src/node_snapshotable.cc

```diff
@@ -181,14 +181,14 @@ class FileIO {
     return name;
   }
 
-  FILE* f = nullptr;
   bool is_debug = false;
 };
 
-class FileReader : public FileIO {
+class SnapshotDeserializer : public SnapshotSerializerDeserializer {
  public:
-  explicit FileReader(FILE* file) : FileIO(file) {}
-  ~FileReader() {}
+  explicit SnapshotDeserializer(const std::vector<char>* s)
```

Comment:
since we’re only reading with memcpy here, this could probably even just be a `string_view`?

Reply:
I used `std::vector` here for symmetry of the serializer API (which uses std::vector to help with the resizing), `std::string_view` also works but it looks a bit...weird?

Reply:
Yeah, I get that they look a bit asymmetric … :) This was just a suggestion, nothing I’m attached to or anything